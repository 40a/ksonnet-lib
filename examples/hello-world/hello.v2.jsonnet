local core = import "../../kube/core.libsonnet";
local kubeUtil = import "../../kube/util.libsonnet";

local container = core.v1.container;
local pod = core.v1.pod;
local deployment = kubeUtil.app.v1beta1.deployment + core.extensions.v1beta1.deployment;

{
  local nginxContainer =
    container.Default("nginx", "nginx:1.7.9") +
    container.NamedPort("http", 80),

  "deployment.json":
    deployment.FromContainer("nginx-deployment", 2, nginxContainer) +
    deployment.mixin.spec.RollingUpdateStrategy() +
    deployment.mixin.spec.Selector({ "app": "nginx" }),
}
