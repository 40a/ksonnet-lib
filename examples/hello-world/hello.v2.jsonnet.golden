{
   "deployment.json": {
      "apiVersion": "extensions/v1beta1",
      "kind": "Deployment",
      "metadata": {
         "labels": { },
         "name": "nginx-deployment"
      },
      "spec": {
         "replicas": 2,
         "selector": {
            "matchLabels": {
               "app": "nginx"
            }
         },
         "strategy": {
            "rollingUpdate": {
               "maxSurge": 1,
               "maxUnavailable": 1
            },
            "type": "RollingUpdate"
         },
         "template": {
            "metadata": {
               "labels": {
                  "app": "nginx"
               }
            },
            "spec": {
               "containers": [
                  {
                     "env": [ ],
                     "image": "nginx:1.7.9",
                     "imagePullPolicy": "Always",
                     "name": "nginx",
                     "ports": [
                        {
                           "containerPort": 80,
                           "name": "http"
                        }
                     ]
                  }
               ],
               "volumes": [ ]
            }
         }
      }
   }
}
